File: examples/python/ds/model/publisher.py
from ds.model._reference import *
from quartz.error import Error
from quartz.util import Parse

#
# Entity
#

class Publisher(PublisherRef):

    def __init__(self, parent_id=None, name=None, id=None):
        super().__init__(id)
        self.parent_id = parent_id
        self.name = name

    def __str__(self):
        return Parse.string(self._name)

    def __repr__(self):
        ref = super().__repr__()
        parent_id = Parse.string(self._parent_id)
        name = Parse.string(self._name)
        return "{}(parent_id:{}, name:{})".format(ref, parent_id, name)

    def _validateRef(self, x):
        if isinstance(x, int) or isinstance(x, PublisherRef):
            return int(x)   

    @property
    def parent_id(self):
        return self._parent_id
    @parent_id.setter
    def parent_id(self, x):
        self._parent_id = PublisherRef.valid(x, True)

    @property
    def name(self):
        return self._name
    @name.setter
    def name(self, x):
        self._name = x

    def copy(self, x):
        if isinstance(x, Publisher):
            super().copy(x)
            self.parent_id = x.parent_id
            self.name = x.name
        return self

File: examples/php/ds/model/publisher.php
<?php
namespace books\model;
require_once __DIR__ . '/../base.php';


class Publisher extends \books\PublisherRef {
    public $parent_id;
    public $name;

    function __construct($parent_id=null, $name=null, $id=null) {
        parent::__construct($id);
        $this->parent_id = $parent_id;
        $this->name = $name;
    }

    public function __toString(): string {
        $ref = parent::__toString();
        $parent = is_null($this->parent_id) ? '∅' : $this->parent_id;
        $name = is_null($this->name) ? '∅' : $this->name;
        return "publisher$ref(parent_id:{$parent}, name:{$name})";
    }

    public function copy($x) {
        if ($x instanceof PublisherRef) {
            parent::copy($x);
            $this->parent_id = x.parent_id;
            $this->name = x.name;
        }
    }
}

?>